Index: lowAltitude_segmentation/Dinov2_segmentation_fine-tuning.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lowAltitude_segmentation/Dinov2_segmentation_fine-tuning.py b/lowAltitude_segmentation/Dinov2_segmentation_fine-tuning.py
new file mode 100644
--- /dev/null	
+++ b/lowAltitude_segmentation/Dinov2_segmentation_fine-tuning.py	
@@ -0,0 +1,30 @@
+"""
+Creating the datasetDict and turning to huggingface dataset,
+Then, fine-tuning that data on Dino-v2 model
+"""
+from datasets import Dataset, DatasetDict, Image
+
+
+image_paths_train = ["path/to/image_1.jpg/jpg", "path/to/image_2.jpg/jpg", ..., "path/to/image_n.jpg/jpg"]
+label_paths_train = ["path/to/annotation_1.png", "path/to/annotation_2.png", ..., "path/to/annotation_n.png"]
+
+def create_dataset(image_paths, label_paths):
+    dataset = Dataset.from_dict({"image": sorted(image_paths),
+                                "label": sorted(label_paths)})
+    dataset = dataset.cast_column("image", Image())
+    dataset = dataset.cast_column("label", Image())
+
+    return dataset
+
+# step 1: create Dataset objects
+train_dataset = create_dataset(image_paths_train, label_paths_train)
+validation_dataset = create_dataset(image_paths_validation, label_paths_validation)
+
+# step 2: create DatasetDict
+dataset = DatasetDict({
+    "train": train_dataset,
+    "validation": validation_dataset,
+  }
+)
+
+
diff --git a/lowAltitude_segmentation/__init__.py b/lowAltitude_segmentation/__init__.py
new file mode 100644
diff --git a/lowAltitude_classification/Dinov2_iNaturalist_classification.py b/lowAltitude_classification/Dinov2_iNaturalist_classification_fine-tuning.py
rename from lowAltitude_classification/Dinov2_iNaturalist_classification.py
rename to lowAltitude_classification/Dinov2_iNaturalist_classification_fine-tuning.py
